src/server/worker.cc

```diff
@@ -564,6 +584,17 @@ void WorkerThread::Start() {
   LOG(INFO) << "[worker] Thread #" << t_.get_id() << " started";
 }
 
+std::map<int, redis::Connection *> Worker::GetConnections() const {
+  std::unique_lock<std::mutex> guard(conns_mu_);
+  std::map<int, redis::Connection *> result;
+  for (auto [fd, _] : conns_) {
```

Comment:
So it's because this traversing uses legacy style traverse, which bring up the problem. The iter would be invalidate when erase happens?

Reply:
Hi @mapleFU ,

Yes, as you commented above, tbb doesn't guarantee the concurrency safe of STL style loop.
I have replaced it with `tbb::parallel_for` and `range()` method.

Best Regards,
Edward